cl/people_db/filters.py

```diff
@@ -249,18 +251,32 @@ class Meta:
         }
 
 
-class PartyFilter(NoEmptyFilterSet):
+class PartyFilter(NoEmptyFilterSet, FilterManyToManyMixin):
     docket = filters.RelatedFilter(
         "cl.search.filters.DocketFilter",
         field_name="dockets",
         queryset=Docket.objects.all(),
+        distinct=True,
     )
     attorney = filters.RelatedFilter(
         "cl.people_db.filters.AttorneyFilter",
         field_name="attorneys",
         queryset=Attorney.objects.all(),
+        distinct=True,
+    )
+    filter_nested_results = filters.BooleanFilter(
+        field_name="roles", method="filter_roles"
     )
 
+    # Attributes for the mixin
+    # **Important:** Keep this mapping up-to-date with any changes to
+    # RelatedFilters or custom labels in this class to avoid unexpected
+    # behavior.
+    join_table_cleanup_mapping = {
+        "attorney__docket": "docket",
+        "attorney__parties_represented": "party",
```

Comment:
Since the filter `"attorney__parties_represented"` is available, would it make sense to add the `party` field to `AttorneyRoleSerializer` to make it easier to understand why those results matched the filter?